File: backend_kapda_junction/src/models/Category.js
const mongoose = require('mongoose');

const categorySchema = new mongoose.Schema({
  name: { type: String, required: true },
  slug: { type: String, unique: true },
  description: String,
  image: String,
  parent: { type: mongoose.Schema.Types.ObjectId, ref: 'Category', default: null },
  metadata: {
    metaTitle: String,
    metaDescription: String,
    keywords: [String]
  },
  sortOrder: { type: Number, default: 0 },
  isActive: { type: Boolean, default: true }
}, { timestamps: true });

categorySchema.index({ parent: 1, isActive: 1 });
categorySchema.pre('save', function (next) {
  if (!this.slug) this.slug = this.name.toLowerCase().replace(/\s+/g, '-').concat('-', Date.now());
  next();
});

module.exports = mongoose.model('Category', categorySchema);

File: flutter_kapda_junction/lib/domain/entities/category.dart
import 'package:equatable/equatable.dart';

class Category extends Equatable {
  final String id;
  final String name;
  final String slug;
  final String? description;
  final String? image;
  final String? parentId;
  final List<Category> children;
  final int sortOrder;
  final bool isActive;

  const Category({
    required this.id,
    required this.name,
    required this.slug,
    this.description,
    this.image,
    this.parentId,
    this.children = const [],
    required this.sortOrder,
    required this.isActive,
  });

  bool get hasChildren => children.isNotEmpty;

  @override
  List<Object?> get props => [id, name, slug];
}
